docs/concept-workflows.md

```diff
@@ -1,30 +1,210 @@
+- **Use-case**:
+
+#### Run timeout
+
+- **Description**: This is the maximum amount of time that a single Workflow run is restricted to.
+- **Use-case**:
+
+#### Task timeout
+
+- **Description**: This is the maximum amount of time that the Server will wait for the Worker to start processing a Task after the Task has been pulled from the Task Queue.
+If a [Command Task](/docs/glossary/#command) is lost, it is retried after this timeout.
+- **Use-case**:
+
+### Retry policy
+
+There may be scenarios where you need to retry a Workflow's execution from the very beginning.
+In this case, you can use supply a retry policy when you start the Workflow.
+
+#### Initial interval
+
+- **Description**: Amount of time that must elapse before the first retry occurs.
+- **Use-case**:
+
+#### Backoff coefficient
+
+- **Description**: Retries occur exponentially.
+The coefficient specifies how fast the retry interval is growing. The coefficient of 1 means that the retry interval is always equal to the `InitialInterval`.
+- **Use-case**:
+
+#### Maximum interval
+
+- **Description**: Specifies the maximum interval between retries.
+Useful for coefficients of more than 1.
+- **Use-case**:
+
+#### Maximum attempts
+
+- **Description**: Specifies the maximum number of attempts that can be made to execute a Workflow in the presence of failures. If this limit is exceeded, the Workflow fails without retrying again.
+- **Use-case**:
+
+#### Non-retryable error reasons
+
+- **Description**: Specifies errors that shouldn't be retried.
+- **Use-case**: There may be errors that you know of that should not trigger a retry.
+In this case you can specify them such that if they occur, the Workflow will not be retried.
+
+### The Task Queue
+
+The only required Workflow options parameter is the name of a [Task Queue](/docs/concept-task-queues).
+Read the [Task Queues concept page](/docs/concept-task-queues) for a better overview.
+But essentially a Task Queue is the mechanism by which any given Worker knows which piece of code to execute next.
+A Workflow can only use one Task Queue, and the from a developer's perspective it is named and managed as a simple string value.
+
+### Custom Id
+
+You may assign a custom Id to a Workflow.
+This Id is meant for business level identification such as a customer Id or an order Id.
+The Temporal Server enforces the uniqueness of the Id, within a [Namespace](/docs/server-namespaces) based on the Workflow Id reuse policy.
+Any attempt to start a Workflow that has the same Id of a Workflow with a reuse policy that does not allow it, is going to fail with a "Workflow execution already started" error.
+
+:::note
+
+A Workflow is uniquely identified by its Namespace, Workflow Id, and Run Id.
+
+:::
+
+#### Allow duplicate failed only policy
+
+- **Description**: Specifying this means that the Workflow is allowed to start only if a previously executed Workflow with the same Id has failed.
+- **Use case**:
```

Comment:
@mfateev ?

Reply:
Use it when there is a need to reexecute a failed workflow and guarantee that succesfully completed workflow will not be reexecuted.